File: microservices.springframework/law-service/src/main/java/swd392/lawservice/infrastructure/configuration/SecurityConfiguration.java
package swd392.lawservice.infrastructure.configuration;

import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.PropertySource;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.web.SecurityFilterChain;
import org.springframework.web.cors.CorsConfiguration;
import org.springframework.web.cors.CorsConfigurationSource;
import org.springframework.web.cors.UrlBasedCorsConfigurationSource;

import java.util.Arrays;
import java.util.List;

@Configuration
@EnableWebSecurity
@RequiredArgsConstructor
@PropertySource("classpath:security.properties")
public class SecurityConfiguration {

    @Value("#{'${app.allowed-origins}'.split(',')}")
    private List<String> ALLOWED_ORIGINS;
    private static final List<String> ALLOWED_METHODS = Arrays.asList("GET", "POST", "PUT", "DELETE");
    private static final List<String> ALLOWED_HEADERS = Arrays.asList("*");
    private static final long CORS_MAX_AGE = 3600L;
    private static final String CORS_PATTERN = "/**";
    private static final String[] OPEN_API = {"/**"};

    @Bean
    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
        http
            .csrf(csrf -> csrf.disable())
            .cors(cors -> cors.configurationSource(corsConfigurationSource()))
            .authorizeHttpRequests(authz -> authz
                .requestMatchers(OPEN_API).permitAll()
                // .anyRequest().authenticated()
            );
        return http.build();
    }

    @Bean
    public CorsConfigurationSource corsConfigurationSource() {
        CorsConfiguration corsConfiguration = new CorsConfiguration();
        corsConfiguration.setAllowedOrigins(ALLOWED_ORIGINS);
        corsConfiguration.setAllowedMethods(ALLOWED_METHODS);
        corsConfiguration.setAllowCredentials(true);
        corsConfiguration.setAllowedHeaders(ALLOWED_HEADERS);
        corsConfiguration.setMaxAge(CORS_MAX_AGE);

        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
        source.registerCorsConfiguration(CORS_PATTERN, corsConfiguration);
        return source;
    }
}

File: microservices.springframework/cloud-storage-service/src/main/kotlin/aas/cloudstorageservice/infrastructure/configuration/SecurityConfiguration.kt
package aas.cloudstorageservice.infrastructure.configuration

import lombok.RequiredArgsConstructor
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.PropertySource
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.web.SecurityFilterChain
import org.springframework.web.cors.CorsConfiguration
import org.springframework.web.cors.CorsConfigurationSource
import org.springframework.web.cors.UrlBasedCorsConfigurationSource

@Configuration
@EnableWebSecurity
@RequiredArgsConstructor
@PropertySource("classpath:security.properties")
class SecurityConfiguration {

    @Value("#{'\${app.allowed-origins}'.split(',')}")
    private val ALLOWED_ORIGINS: MutableList<String?>? = null
    private val ALLOWED_METHODS = listOf("GET", "POST", "PUT", "DELETE");
    private val ALLOWED_HEADERS = listOf("*");
    private val CORS_MAX_AGE: Long = 3600;
    private val CORS_PATTERN = "/**";
    private val OPEN_API = arrayOf("/**");

    @Bean
    fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
        http
            .csrf { it.disable() }
            .cors { it.configurationSource(corsConfigurationSource()) }
            .authorizeHttpRequests { it
                .requestMatchers(*OPEN_API).permitAll()
                // .anyRequest().authenticated()
            }
        return http.build();
    }

    @Bean
    fun corsConfigurationSource(): CorsConfigurationSource {
        val corsConfiguration = CorsConfiguration();
        corsConfiguration.allowedOrigins = ALLOWED_ORIGINS;
        corsConfiguration.allowedMethods = ALLOWED_METHODS;
        corsConfiguration.allowCredentials = true;
        corsConfiguration.allowedHeaders = ALLOWED_HEADERS;
        corsConfiguration.maxAge = CORS_MAX_AGE;
        val source = UrlBasedCorsConfigurationSource();
        source.registerCorsConfiguration(CORS_PATTERN, corsConfiguration);
        return source;
    }
}